db/data/20200727124151_add_transparency_identifiers_to_all_activities.rb

```diff
@@ -0,0 +1,19 @@
+class AddTransparencyIdentifiersToAllActivities < ActiveRecord::Migration[6.0]
+  def up
```

Comment:
As @lozette and I discussed, it is unlikely anything else will modify this data while this transaction runs. If there _is_ anything else that could modify the table, then the `transparency_identifier` field or something it's derived from could change between all the records being loaded in `Activity.all.each`, and each iteration of the loop.

Using a compare-and-swap in the loop body would make sure this only updates records where `transparency_identifier` is still blank:

```rb
      transparency_identifier = activity.iati_identifier
      Activity.where(id: activity.id, transparency_identifier: nil).update_all(transparency_identifier: transparency_identifier)
```

Reply:
I should add the nice thing about this implementation is it's safe if it partially fails -- if some of the records are assigned an ID and then the migration crashes, it can safely be resumed :)